zerver/views/users.py

```diff
@@ -315,7 +315,7 @@ def update_user_backend(
         validate_user_custom_profile_data(
             target.realm.id, clean_profile_data, acting_user=user_profile
         )
-        do_update_user_custom_profile_data_if_changed(target, clean_profile_data)
+        do_update_user_custom_profile_data_if_changed(target, clean_profile_data, user_profile)
```

Comment:
I guess we should also send notification message when user removes a custom profile field value.